libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -160,10 +206,76 @@ func scanExistingSegments(path string) ([]*queueSegment, error) {
 }
 
 func (segment *queueSegment) sizeOnDisk() uint64 {
-	return uint64(segment.endOffset) + segmentHeaderSize
+	var headerSize int
+	if segment.header != nil {
+		headerSize = segment.header.sizeOnDisk()
+	} else {
+		headerSize = segmentHeaderSize
+	}
+	return uint64(segment.endOffset) + uint64(headerSize)
```

Comment:
I wonder if it would be easier to just track absolute file offsets in general. Then we care about the header size only when opening a file. The file format is quite simple, do we really need to separate by data region?

Reply:
I wonder this also, having worked through the consequences of a schema update here... I'll see how much work this would be.

Reply:
It turns out it was a fair amount of work, but this has been revived and reworked to use absolute file positions everywhere and convert offsets from older schema versions on load.